www/scripts/login/login.js

```diff
@@ -0,0 +1,219 @@
+      });
+
+      this.txtUser = new TextBox({
+        class : 'form-input',
+        name  : 'username'
+      });
+
+      this.txtPass = new TextBox({
+        class : 'form-input',
+        name  : 'password',
+        type  : 'password'
+      });
+
+      var that = this;
+      this.btnSubmit = new Button({
+        label   : "Login",
+        onClick : function () {
+          that._doLogin();
+        }
+      });
+    },
+
+    postCreate : function() {
+      this.addChild(this._mbox);
+      this._mbox.hide();
+
+      var authParams = CC_AUTH_SERVICE.getAuthParameters();
+
+      if (!authParams.requiresAuthentication || authParams.sessionStillActive) {
+        domClass.add(this._mbox.domNode, 'mbox-success');
+
+        var message = '';
+        if (!authParams.requiresAuthentication)
+          message = "This server allows anonymous access.";
+        else if (authParams.sessionStillActive)
+          message = "You are already logged in.";
+        this._mbox.show("No authentication required.", message);
+
+        var returnTo = hash.getState('returnto') || '';
+        window.location = window.location.origin + '/' + returnTo;
+      } else {
+        var authMethods = CC_AUTH_SERVICE.getAcceptedAuthMethods();
+        if (!authMethods.includes('Username:Password')) {
+          domClass.add(this._mbox.domNode, 'mbox-error');
+          this._mbox.show("Server rejects username-password authentication!",
+                          "This login form cannot be used.");
+        } else {
+          var cntPrompt = domConstruct.create('div', {
+            class : 'formElement'
+          }, this.containerNode);
+          var lblPrompt = domConstruct.create('span', {
+            class     : 'login-prompt',
+            style     : 'width: 100%',
+            innerHTML : this.loginPrompt
+          }, cntPrompt);
+
+          // Render the login dialog's controls.
+
+          var cntUser = domConstruct.create('div', {
+            class : 'formElement'
+          }, this.containerNode);
+          var lblUser = domConstruct.create('label', {
+            class     : 'formLabel bold',
+            innerHTML : "Username: ",
+            for       : 'username'
+          }, cntUser);
+          domConstruct.place(this.txtUser.domNode, cntUser);
+
+          var cntPass = domConstruct.create('div', {
+            class : 'formElement'
+          }, this.containerNode);
+          var lblPass = domConstruct.create('label', {
+            class     : 'formLabel bold',
+            innerHTML : "Password: ",
+            for       : 'password'
+          }, cntPass);
+          domConstruct.place(this.txtPass.domNode, cntPass);
+
+          var cntLogin = domConstruct.create('div', {
+            class : 'formElement'
+          }, this.containerNode);
+          domConstruct.place(this.btnSubmit.domNode, cntLogin);
+
+          var that = this;
+          function keypressHandler(evt) {
+            if (evt.keyCode === keys.ENTER) {
```

Comment:
Problem on login:
- Type the username
- Type the password
- Hit enter

The login failed because the password wasn't set properly:
```bash
[1, "performLogin", 1, 0, {1: {str: "Username:Password"}, 2: {str: "test:"}}]
```

Reply:
Fixed.